### PR TITLE
feat:  extend tracking performance scripts

### PR DESCRIPTION
**Pull request by asalzburger**
This PR:
 * extends the functionality of `Examples/Scripts/TrackingPerformance/boundParamResiduals`,
 * creates a compiled standalone version as `ActsExampleResidualAndPulls`
 * adds `ACTS_BUILD_EXAMPLES_SCRIPTS` CMake flag

It also fixes the issue of `phi` residuals that did not respect periodic scaling.

### CODE REVIEW

**Comment by asalzburger** [comment]
phi_pull

FYI: Phi-pulls before the fix.

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C [hunk 1]
```diff
@@ -0,0 +1,852 @@
...
+
+  /// Helper method to make a volLayId string for identification & cut
+  ///
+  /// ensures a unique identification
+  auto volLayIdCut = [](int vol, int lay) -> std::array<std::string, 2> {
+    if (vol < 0 and lay < 0) {
+      return {std::string("all_"), ""};
+    }
+
+    std::string vlstr = "vol" + std::to_string(vol);
```
Comment (XiaocongAi) [inline Examples/Scripts/TrackingPerformance/boundParamResolution.C:370]: It seems like we should have the volume id cut only if `vol > 0`, and have the layer id cut only if `vol > 0 and layer > 0`, e.g.  an input of `vol =  -1` and `lay = 8` can mean that no cut is necessary ?

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+        setHistStyle(res_smt[id_par], 4);
+
+        setHistStyle(pull_prt[id_par], 1);
+        setHistStyle(pull_flt[id_par], 2);
+        setHistStyle(pull_smt[id_par], 4);
+      }
+    }
+  }
+
+  // Section: Histogram filling
```
Comment (XiaocongAi) [inline Examples/Scripts/TrackingPerformance/boundParamResolution.C:559]:
```suggestion
  // Section 4: Histogram filling
```

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C [hunk 3]
```diff
@@ -0,0 +1,852 @@
...
+          pull_smt[vlID + paramNames[2]]->Fill(pull_PHI_smt->at(i), 1);
+          pull_smt[vlID + paramNames[3]]->Fill(pull_THETA_smt->at(i), 1);
+          pull_smt[vlID + paramNames[4]]->Fill(pull_QOP_smt->at(i), 1);
+          pull_smt[vlID + paramNames[5]]->Fill(pull_T_smt->at(i), 1);
+        }
+      }
+    }
+  }
+
+  // Section: Histogram plotting
```
Comment (XiaocongAi) [inline Examples/Scripts/TrackingPerformance/boundParamResolution.C:681]:
```suggestion
  // Section 5: Histogram plotting
```

**Review by XiaocongAi — COMMENTED** [review]
Hi @asalzburger , thank you for this really nice PR. I have only a few minor comments.

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C [hunk 4]
```diff
@@ -0,0 +1,852 @@
...
+  double green[number] = {0., 1., 1., 1., 0.};
+  double blue[number] = {1., 1., 0., 0., 0.};
+  double stops[number] = {0., rel_good - rel_window, rel_good,
+                          rel_good + rel_window, 1.};
+  h->SetContour(n);
+
+  TColor::CreateGradientColorTable(number, stops, red, green, blue, n);
+}
+
+// loc1, phi, theta, q/p, t) at all track states from root file produced by the
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Examples/Scripts/TrackingPerformance/boundParamResolution.C:70]: Added '('

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C (hunk 1 above)
Comment (asalzburger) [reply Examples/Scripts/TrackingPerformance/boundParamResolution.C:370]: if `volume_id < 0` then necessarily `layer_id < 0`, hence this case is covered by

```c++
    if (vol < 0 and lay < 0) {
      return {std::string("all_"), ""};
    }
```

In L366.

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C (hunk 2 above)
Comment (asalzburger) [reply Examples/Scripts/TrackingPerformance/boundParamResolution.C:559]: Fixed.

File: Examples/Scripts/TrackingPerformance/boundParamResolution.C (hunk 3 above)
Comment (asalzburger) [reply Examples/Scripts/TrackingPerformance/boundParamResolution.C:681]: Also fixed.

**Comment by asalzburger** [comment]
Hi @XiaocongAi  - I addressed the PR comments, please have another look if you're ok with that. I might extend the functionality of this script in the future, but I think that can go into another PR.

**Comment by asalzburger** [comment]
I've resolved the merge conflict, will merge when CI runs through.

**Pull request merged**